deploy/osps/default/osp-amzn2.yaml

```diff
@@ -81,7 +81,15 @@ spec:
 
               curl -s -k -v --header 'Authorization: Bearer {{ .Token }}' {{ .ServerURL }}/api/v1/namespaces/cloud-init-settings/secrets/{{ .SecretName }} | jq '.data["cloud-config"]' -r| base64 -d > /etc/cloud/cloud.cfg.d/{{ .SecretName }}.cfg
               cloud-init clean
-              cloud-init --file /etc/cloud/cloud.cfg.d/{{ .SecretName }}.cfg init
+              CLOUD_INIT_VERSION=$(cloud-init --version | awk '{print $2}')
+              # Compare the semver values of cloud-init versions to determine the correct command to run.
+              # This is required because the command line arguments for cloud-init changed in version 24.0.0.
+              if [[ $(echo -e "24.0.0\n$CLOUD_INIT_VERSION" | sort -V | head -n1) = "24.0.0" ]]; then
```

Comment:
24.0.0 doesn't exist, so this check is fine, but can we ensure that the comment is correct and references 24.1 instead of 24.0.0?

Reply:
Sure, updated